### PR TITLE
HPCC-28834 Roxie UdpReceiver socket not open msg repeats forever

### PR DESCRIPTION
**Pull request by richardkchapman**
Slow down rate of logging

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by richardkchapman** [comment]
Might want to also terminate after N minutes of continuous failure (where N might be 1)

**Review by ghalliday — APPROVED** [review]

File: roxie/udplib/udptrr.cpp [hunk 1]
```diff
@@ -1223,14 +1225,30 @@ class CReceiveManager : implements IReceiveManager, public CInterface
...
                     {
-                        StringBuffer s;
-                        DBGLOG("UdpReceiver: failed %i %s", flow_port, e->errorMessage(s).str());
+                        unsigned now = msTick();
+                        if (lastReport-now < 60000)
+                            suppressed++;
+                        else
+                        {
+                            StringBuffer s;
+                            DBGLOG("UdpReceiver: failed %d times %i %s", suppressed+1, flow_port, e->errorMessage(s).str());
```
Comment (mckellyln) [inline roxie/udplib/udptrr.cpp:1234]: Just to pick, %u for unsigned ?

File: roxie/udplib/udptrr.cpp [hunk 2]
```diff
@@ -1200,6 +1200,8 @@ class CReceiveManager : implements IReceiveManager, public CInterface
         #endif
             UdpRequestToSendMsg msg;
             unsigned timeout = 5000;
+            unsigned lastReport = 0;
```
Comment (mckellyln) [inline roxie/udplib/udptrr.cpp:1203]: Should we init this to msTick() ?  But certainly no big deal.

**Review by mckellyln — COMMENTED** [review]
Very minor comments, and I am ok with it as is.

File: roxie/udplib/udptrr.cpp (hunk 2 above)
Comment (richardkchapman) [reply roxie/udplib/udptrr.cpp:1203]: No, I want the first time to come out straight away rather than be delayed for a minute

**Comment by mckellyln** [comment]
Maybe and depending on the error code - is this a fatal error and roxie should stop/restart ?
Or after 10000 of these with a specific error code ?
I mean when it happens here it never fixes itself and I think roxie must be manually restarted.

**Comment by ghalliday** [comment]
@richardkchapman I think create a separate PR for that issue.

**Pull request merged**